tools/clang/tools/dxcvalidator/dxcvalidator.cpp

```diff
@@ -73,42 +110,69 @@ static uint32_t runValidation(
 
   if (!Module) {
     DXASSERT_NOMSG(DebugModule == nullptr);
-    if (Flags & DxcValidatorFlags_ModuleOnly) {
+    if (Flags & DxcValidatorFlags_ModuleOnly)
       return ValidateDxilBitcode((const char *)Shader->GetBufferPointer(),
                                  (uint32_t)Shader->GetBufferSize(), DiagStream);
-    } else {
+    else
       return ValidateDxilContainer(Shader->GetBufferPointer(),
                                    Shader->GetBufferSize(), DiagStream);
-    }
   }
 
   llvm::DiagnosticPrinterRawOStream DiagPrinter(DiagStream);
   PrintDiagnosticContext DiagContext(DiagPrinter);
   DiagRestore DR(Module->getContext(), &DiagContext);
 
-  HRESULT hr = hlsl::ValidateDxilModule(Module, DebugModule);
-  if (FAILED(hr))
-    return hr;
+  HRESULT HR = hlsl::ValidateDxilModule(Module, DebugModule);
+  if (FAILED(HR))
+    return HR;
   if (!(Flags & DxcValidatorFlags_ModuleOnly)) {
-    hr = ValidateDxilContainerParts(
+    HR = ValidateDxilContainerParts(
         Module, DebugModule,
         IsDxilContainerLike(Shader->GetBufferPointer(),
                             Shader->GetBufferSize()),
         (uint32_t)Shader->GetBufferSize());
-    if (FAILED(hr))
-      return hr;
+    if (FAILED(HR))
+      return HR;
   }
 
-  if (DiagContext.HasErrors() || DiagContext.HasWarnings()) {
+  if (DiagContext.HasErrors() || DiagContext.HasWarnings())
     return DXC_E_IR_VERIFICATION_FAILED;
-  }
 
   return S_OK;
 }
 
-static uint32_t
-runRootSignatureValidation(IDxcBlob *Shader,
-                           AbstractMemoryStream *DiagMemStream) {
+static uint32_t runValidation(IDxcBlob *Shader,
```

Comment:
I think it would be better if this was combined with the other overload, and the combined function handled whether `DxcValidatorFlags_ModuleOnly` was being performed, or if we have a full container to sign.  Right now it's confusing, and an internal path with a debug module can't go through the signing path.

Reply:
This can be accomplished by moving debug module serialization out, but it will also require a change to `ValidateDxilContainer` to take the `llvm::Module*` instead of the serialized bitcode.

Reply:
Hmm... perhaps this isn't sufficient, since it might need to load the debug module from the container instead.  So maybe we just need to add a debug `llvm::Module` argument to pass along if we already have one.